### PR TITLE
Fix/5837

### LINKED ISSUE #5837: Long query names prevent the user from accessing the overflow menu
# Bug report

## Describe the bug

In the SQL editor, queries with long names cannot be renamed or deleted because the button that brings up the overflow menu with these actions is pushed outside of the container

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Go to `https://app.supabase.io/project/<project-id>/sql`
2. Click on "New Query"
3. Create a query with a name "This query deliberately has a long name to be descriptive"
4. Try to rename it

## Expected behavior

The overflow menu button should remain visible and accessible, no matter how long the query name is.
It would also be helpful to set a `title` attribute to allow the user to see the full name of the query on long hover.

## Screenshots

<details><summary>Click to see</summary>

long-query-name-full-page
long-query-name-inspected

</details>

## System information

- OS: Windows
- Browser (if applies): Latest Chrome
- Version of supabase-js: N/A
- Version of Node.js: N/A

### PR DESCRIPTION
**Pull request by joshenlim**
Resolves https://github.com/supabase/supabase/issues/5837

### CODE REVIEW

File: studio/components/ui/ProductMenu/ProductMenuItem.tsx [hunk 1]
```diff
@@ -25,8 +25,13 @@ const ProductMenuItem: FC<Props> = ({
 }) => {
   const menuItem = (
     <Menu.Item icon={icon} rounded active={isActive} onClick={onClick}>
-      <div className="flex w-full justify-between items-center">
-        <span className={'truncate flex items-center ' + textClassName}>{name}</span>
+      <div className="flex w-full items-center justify-between">
+        <span
+          title={typeof name === 'string' ? name : ''}
```
[the remark this answers is not in the document]
Comment (MildTomato) [reply studio/components/ui/ProductMenu/ProductMenuItem.tsx:30]: ahh, Menu.Item should really be accepting props like `title`.

**Review by MildTomato — COMMENTED** [review]
.

File: studio/components/ui/ProductMenu/ProductMenuItem.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshenlim) [reply studio/components/ui/ProductMenu/ProductMenuItem.tsx:30]: its cause name could also be a ReactNode - ref `StorageMenu` where we also show the `public` badge

File: studio/components/layouts/SQLEditorLayout/SQLEditorMenu.tsx [hunk 2]
```diff
@@ -1,217 +1,277 @@
...
-}) => {
-  const { ui, content } = useStore()
-  const router = useRouter()
-  const { queryId } = router.query
+import RenameQuery from 'components/to-be-cleaned/SqlEditor/RenameQuery'
+import { createSqlSnippet } from 'components/to-be-cleaned/SqlEditor/SqlEditor.utils'
+import ConfirmationModal from 'components/ui/ConfirmationModal'
+import ProductMenuItem from 'components/ui/ProductMenu/ProductMenuItem'
+
+const OpenQueryItem = observer(({ tabInfo }: { tabInfo: any }) => {
```
Comment (phamhieu) [inline studio/components/layouts/SQLEditorLayout/SQLEditorMenu.tsx:30]: proper type

File: studio/components/layouts/SQLEditorLayout/SQLEditorMenu.tsx [hunk 3]
```diff
@@ -1,217 +1,277 @@
...
+      isActive={active}
+      name={name}
+      action={active && <DropdownMenu tabInfo={tabInfo} />}
+      onClick={() => sqlEditorStore.selectTab(id)}
+      textClassName="w-44"
+    />
   )
+})
+
+const DropdownMenu = observer(({ tabInfo }: { tabInfo: any }) => {
```
Comment (phamhieu) [inline studio/components/layouts/SQLEditorLayout/SQLEditorMenu.tsx:47]: ditto

File: studio/components/layouts/SQLEditorLayout/SQLEditorMenu.tsx [hunk 4]
```diff
@@ -1,217 +1,277 @@
...
+        onComplete={onCloseRenameModal}
+      />
+
+      <ConfirmationModal
+        header="Confirm to remove"
+        buttonLabel="Confirm"
+        visible={deleteModalOpen}
+        onSelectConfirm={async () => {
+          await contentStore.del(id)
+          await contentStore.load()
```
Comment (phamhieu) [inline studio/components/layouts/SQLEditorLayout/SQLEditorMenu.tsx:112]: look strange. i will take a look at the content and sql store again

**Review by phamhieu — APPROVED** [review]

**Pull request merged**